zipkin-thrift/src/main/thrift/com/twitter/zipkin/zipkinCore.thrift

```diff
@@ -15,18 +15,118 @@ namespace java com.twitter.zipkin.thriftjava
+ */
 const string CLIENT_RECV = "cr"
+/*
+ * The server sent ("ss") a response to a client. There is only one response
+ * per span. If there's a transport error, each attempt can be logged as a
+ * WIRE_SEND annotation.
+ *
+ * Typically, a trace ends with a server send, so the last timestamp of a trace
+ * is often the timestamp of the root span's server send.
+ *
+ * If chunking is involved, each chunk could be logged as a separate
+ * SERVER_SEND_FRAGMENT in the same span.
+ *
+ * Annotation.host is not the client. It is the host which logged the send
+ * event, almost always the server.
+ *
+ * To record the actual endpoints of the client and server side of the span,
+ * use CLIENT_ADDR and SERVER_ADDR respectively.
+ */
 const string SERVER_SEND = "ss"
+/*
+ * The server received ("cr") a request from a client. There is only one
+ * request per span.  For example, if duplicate responses were received, each
+ * can be logged as a WIRE_RECV annotation.
+ *
+ * Typically, a trace starts with a server receive, so the first timestamp of a
+ * trace is often the timestamp of the root span's server receive.
+ *
+ * If chunking is involved, each chunk could be logged as a separate
+ * SERVER_RECV_FRAGMENT in the same span.
+ *
+ * Annotation.host is not the client. It is the host which logged the send
+ * event, almost always the server.
+ *
+ * To record the actual endpoints of the client and server side of the span,
+ * use CLIENT_ADDR and SERVER_ADDR respectively.
+ */
 const string SERVER_RECV = "sr"
+/*
+ * Optionally logs an attempt to send a message on the wire. Multiple wire send
+ * events could indicate network retries. A lag between client or server send
+ * and wire send might indicate queing or processing delay.
+ */
 const string WIRE_SEND = "ws"
+/*
+ * Optionally logs an attempt to receive a message from the wire. Multiple wire
+ * receive events could indicate network retries. A lag between wire receive
+ * and client or server receive might indicate queing or processing delay.
+ */
 const string WIRE_RECV = "wr"
+/*
+ * Optionally logs progress of a (CLIENT_SEND, WIRE_SEND). For example, this
+ * could be one chunk in a chunked request.
+ */
+const string CLIENT_SEND_FRAGMENT = "csf"
+/*
+ * Optionally logs progress of a (CLIENT_RECV, WIRE_RECV). For example, this
+ * could be one chunk in a chunked response.
+ */
+const string CLIENT_RECV_FRAGMENT = "crf"
+/*
+ * Optionally logs progress of a (SERVER_SEND, WIRE_SEND). For example, this
+ * could be one chunk in a chunked response.
+ */
+const string SERVER_SEND_FRAGMENT = "ssf"
+/*
+ * Optionally logs progress of a (SERVER_RECV, WIRE_RECV). For example, this
+ * could be one chunk in a chunked request.
+ */
+const string SERVER_RECV_FRAGMENT = "srf"
 
-#************** Common binary annotation keys **************
-// The endpoint associated with CLIENT_ annotations is not necessarily Annotation.host
+#***** BinaryAnnotation.key where value = [1] and annotation_type = BOOL ******
```

Comment:
In zipkin all annotations/binary annotations log the "local" ip/port in the host field.

This allows you to group events by the host creating them (via endpoint.ip). For example, if I have a host that's acting awry, I could narrow that down and see all activity on that host. If we didn't store the local IP in annotation.host, we could not do this.

By using the local ip/port in annotations (log statements), we can also see when a client and server disagree on a value of the same key. For example, if "http.uri" can be logged as "/v1/foo" on the client, and received as "/foo" on the server. If we logged the server's ip as the host of a "http.uri" annotation sent by a client, we wouldn't be see differences like this, or know which was caused by who.

For better or worse, there are two exceptions to the "always use local ip/port in annotation/binary annotations" rule:  `CLIENT_ADDR` and `SERVER_ADDR`. When a binary annotation key is "ca" or "sa", the value of `[1]` is ignored, as the actual value is the host/endpoint. These special annotations allow us to make visible the ip and port of uninstrumented clients or servers. For example, a root span begins with a "sr", but we often can tell what the IP of the client was. For example, we could store the `X-Forwarded-For` value. When sending a call to an uninstrumented destination, such as an Amazon S3 bucket, we can record the ip/port as `SERVER_ADDR`. 

What's not immediately clear is why we put `[1]` as a value, instead of serializing the endpoint in some encoding. The answer is more than just avoiding duplicating binaryAnnotation.host as also binaryAnnotation.value. It is that binaryAnnotation.host is already destructured and indexed in all backends. If we stored the host as binaryAnnotation.value, we would need to do heavy lifting to parse that across potentially large amounts of data.

This optimization comes at a cost of intuitiveness, as the value isn't where we expect it to be. It also requires guidance of log SERVER_ADDR from the client and visa-versa, as without that, interpretation is confusing as well. Theses issues may lead to refactoring in [zipkin v2](https://docs.google.com/document/d/1ixxEs9TvhiGjJObGbRSPhSna3zHdadoUTQIZ5JKgLzU/edit).